docs/Queries.md

```diff
@@ -0,0 +1,215 @@
+# Query examples
+
+This is a (non-exhaustive) list of possible queries 
+and their responses for `indexer-service-rs`.
+
+## Supported request and response format examples
+
+```bash
+curl http://localhost:7600/
+```
+
+```
+Service is up and running
+```
+
+```bash
+curl http://localhost:7600/version
+```
+
+```json
+{ "version":"0.1.0", "dependencies": {..} }
+```
+
+```bash
+curl http://localhost:7600/info
+```
+
+```json
+{ "publicKey": "0xacb05407d78129b5717bb51712d3e23a78a10929" }
+```
+
+# Subgraph queries
+
+## Checks for receipts and authorization
+```bash
+curl -X POST \
+  -H 'Content-Type: application/json' \
+  -H 'Authorization: Bearer token-for-graph-node-query-endpoint' \
+  --data '{"query": "{_meta{block{number}}}"}' \
+  http://localhost:7600/subgraphs/id/QmacQnSgia4iDPWHpeY6aWxesRFdb8o5DKZUx96zZqEWrB
+```
+```json
+{
+    "attestable": true,
+    "graphQLResponse": "{\"data\":{\"_meta\":{\"block\":{\"number\":10666745}}}}"
+}
+```
+## Takes hex representation for subgraphs deployment id aside from IPFS hash representation
+```bash
+curl -X POST \
+  -H 'Content-Type: application/json' \
+  -H 'Authorization: Bearer token-for-graph-node-query-endpoint' \
+  --data '{"query": "{_meta{block{number}}}"}' \
+  http://localhost:7600/subgraphs/id/0xb655ca6f49e73728a102219726ff678d61d8fb792874792e9f0d9887dc616600
+```
+```json
+{
+    "attestable": true,
+    "graphQLResponse": "{\"data\":{\"_meta\":{\"block\":{\"number\":10666745}}}}"
+}
+```
+
+## Free query auth token check failed
+
+```bash
+curl -X POST \
+  -H 'Content-Type: application/json' \
+  -H 'Authorization: blah' \
+  --data '{"query": "{_meta{block{number}}}"}' \
+  http://localhost:7600/subgraphs/id/0xb655ca6f49e73728a102219726ff678d61d8fb792874792e9f0d9887dc616600
+```
+
+```json
+{
+  "message":"No valid receipt or free query auth token provided"
+}
+```
+
+## Subgraph health check
+```bash
+curl http://localhost:7600/subgraphs/health/QmVhiE4nax9i86UBnBmQCYDzvjWuwHShYh7aspGPQhU5Sj
+```
+```json
+{
+    "health": "healthy"
+}
+```
+## Unfound subgraph
+```bash
+curl http://localhost:7600/subgraphs/health/QmacQnSgia4iDPWHpeY6aWxesRFdb8o5DKZUx96zZqEWrB
+```
+```json
+{
+    "error": "Deployment not found"
+}
+```
+## Failed Subgraph
+```bash
```

Comment:
```suggestion

```bash
curl http://localhost:7600/subgraphs/health/QmVhiE4nax9i86UBnBmQCYDzvjWuwHShYh7aspGPQhU5Sj
```

```json
{
    "health": "healthy"
}
```

## Unfound subgraph

```bash
curl http://localhost:7600/subgraphs/health/QmacQnSgia4iDPWHpeY6aWxesRFdb8o5DKZUx96zZqEWrB
```

```json
{
    "error": "Deployment not found"
}
```

## Failed Subgraph

```bash
```